app/components/certificate/certificate-display.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Flex,
+  Box,
+  HStack,
+  Text,
+  Heading,
+  Tooltip,
+  IconButton,
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionIcon,
+  AccordionPanel,
+  useToast,
+} from '@chakra-ui/react';
+import { DownloadIcon, CopyIcon } from '@chakra-ui/icons';
+import type { Certificate } from '@prisma/client';
+
+interface Props {
+  cert: Certificate;
+  priv: Boolean;
+}
+
+export default function CertificateDisplay(props: Props) {
+  const toast = useToast();
+
+  function onCopy() {
+    let key = props.cert.certificate;
+    let val = 'Public Key';
+
+    if (props.priv) {
+      key = props.cert.privateKey;
+      val = 'Private Key';
+    }
+
+    navigator.clipboard.writeText(key);
+    toast({
+      title: `${val} was copied to clipboard`,
+      position: 'bottom-right',
+      status: 'success',
+    });
+  }
+
+  return (
+    <Flex flexDirection="column" gap="5">
+      <Flex flexDirection="column" gap="4">
+        <HStack gap="4">
+          <Heading as="h4" size="sm">
+            {props.priv ? 'Private Key' : 'Public Key'}
+          </Heading>
+          <Tooltip label="Copy Public Key">
+            <IconButton
+              backgroundColor="transparent"
+              color="black"
+              size="xs"
+              _hover={{
+                background: 'whitesmoke',
+                color: 'teal.500',
+              }}
+              aria-label="Copy to Clipboard"
+              icon={<CopyIcon fontSize="md" />}
+              onClick={() => onCopy()}
+            />
+          </Tooltip>
+          <Tooltip label="Download Public Key">
```

Comment:
and here

Reply:
Maybe I can pass a string on the prop, and use that as a state of either 'Private' and 'Public' to be used on the text? Would that work?